be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -956,10 +956,44 @@ Status HashJoinNode::open(RuntimeState* state) {
         RETURN_IF_ERROR((*_vother_join_conjunct_ptr)->open(state));
     }
 
-    RETURN_IF_ERROR(_hash_table_build(state));
-    RETURN_IF_ERROR(child(0)->open(state));
+    std::promise<Status> thread_status;
+    std::thread(bind(&HashJoinNode::_hash_table_build_thread, this, state, &thread_status))
+            .detach();
+
+    if (_runtime_filter_descs.empty()) {
+        // Open the probe-side child so that it may perform any initialisation in parallel.
+        // Don't exit even if we see an error, we still need to wait for the build thread
+        // to finish.
+        // ISSUE-1247, check open_status after buildThread execute.
```

Comment:
std::promise<Status> thread_status;
    std::thread(bind(&HashJoinNode::_hash_table_build_thread, this, state, &thread_status))
            .detach();